File: submissions/cpp/238.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int> &nums) {
        vector<int> ret(nums.size(), 1);
        int temp = 1;
        for (int i = 0; i < nums.size() - 1; ++i)
            ret[i + 1] = ret[i] * nums[i];
        for (int i = nums.size() - 2; i >= 0; --i) {
            temp *= nums[i + 1];
            ret[i] *= temp;
        }
        return ret;
    }
};

File: submissions/python3/238.py
class Solution:
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        temp = 1
        ret = [1] * len(nums)
        for i in range(len(nums) - 1):
            ret[i+1] = ret[i] * nums[i]
        for i in range(len(nums)-2, -1, -1):
            temp *= nums[i + 1]
            ret[i] *= temp
        return ret
